src/pages/OfferNew/Introduction/Sidebar/DetailsModal/index.tsx

```diff
@@ -0,0 +1,425 @@
+                        render={(arrayHelpers) => (
+                          <>
+                            <ContentColumnTitle>
+                              {textKeys.DETAILS_MODULE_EXTRABUILDINGS_TABLE_TITLE()}
+                              <ContentColumnTitleButton
+                                type="button"
+                                onClick={() => {
+                                  const defaultExtraBuilding: ExtraBuildingInput = {
+                                    type: ExtraBuildingType.Garage,
+                                    area: 10,
+                                    hasWaterConnected: false,
+                                  }
+                                  arrayHelpers.insert(0, defaultExtraBuilding)
+                                }}
+                              >
+                                {textKeys.DETAILS_MODULE_EXTRABUILDINGS_TABLE_BUTTON()}
+                              </ContentColumnTitleButton>
+                            </ContentColumnTitle>
+
+                            {formikProps.values.house?.extraBuildings?.map(
+                              (_, index) => (
+                                <InputGroup key={index}>
+                                  <DetailInput
+                                    field={
+                                      fieldSchema.house.extraBuildings.type
+                                    }
+                                    formikProps={formikProps}
+                                    nameRoot="house"
+                                    name={`extraBuildings.${index}.type`}
+                                  />
+
+                                  <DetailInput
+                                    field={
+                                      fieldSchema.house.extraBuildings.area
+                                    }
+                                    formikProps={formikProps}
+                                    nameRoot="house"
+                                    name={`extraBuildings.${index}.area`}
+                                  />
+
+                                  <DetailInput
+                                    field={
+                                      fieldSchema.house.extraBuildings
+                                        .hasWaterConnected
+                                    }
+                                    formikProps={formikProps}
+                                    nameRoot="house"
+                                    name={`extraBuildings.${index}.hasWaterConnected`}
+                                  />
+
+                                  <InputGroupDeleteButton
+                                    type="button"
+                                    onClick={() => {
+                                      const isLastItem =
+                                        formikProps.values.house?.extraBuildings
+                                          ?.length === 1
```

Comment:
i think i don't really understand this logic, what if you have 3 extra buildings? then `isLastItem` doesn't make sense for `extraBulidings.length === 1` does it? Or do you mean that this is the only item in the list? Then i think we should specify that as a name instead